odl/operator/operator.py

```diff
@@ -240,7 +240,7 @@ def __init__(self, domain, range, linear=False):
         self._is_linear = bool(linear)
         self._is_functional = isinstance(range, Field)
 
-        if self.is_linear:
+        if self._is_linear:
```

Comment:
Before I saw it in action, I didn't even know that it was possible to use the property at that point. But why shouldn't it? The instance is already created at that point, so the property exists and works once the "private" attribute is in place.
So no, it isn't an optimization, more something like "I just set the private attribute so I know I can use that, not so sure about the public property." I'll change it back and (incrementally) also in other places where I encounter similar code.